chef/cookbooks/neutron/recipes/server.rb

```diff
@@ -149,26 +149,26 @@
 num_vlans = node[:neutron][:num_vlans]
 vlan_end = [vlan_start + num_vlans - 1, 4094].min
 
-if node[:neutron][:networking_plugin] == "cisco"
-  mechanism_driver = "openvswitch,cisco_nexus"
-else
-  mechanism_driver = node[:neutron][:networking_plugin]
-end
-
 directory "/etc/neutron/plugins/ml2" do
   mode 0755
   action :create
   only_if { node[:platform] == "ubuntu" }
 end
 
+# NOTE(toabctl): tenant_network types should have as first element 'ml2_type_drivers_default_tenant_network' and then the rest of the selected type drivers.
+#                when creating tenant networks, it's not possible to manually select the network type. Neutron just tries every selected type until success
+#                so the order is important
+tenant_network_types = [node[:neutron][:ml2_type_drivers_default_tenant_network]] + node[:neutron][:ml2_type_drivers].collect{ |el| el if el != node[:neutron][:ml2_type_drivers_default_tenant_network]}.compact
```

Comment:
Done in https://github.com/crowbar/barclamp-neutron/commit/1504fd042f9cf6bf1e7665d354e3e84e141adf39